chia/wallet/wallet_node.py

```diff
@@ -303,8 +302,6 @@ def _close(self):
             self._process_new_subscriptions_task.cancel()
         if self._retry_failed_states_task is not None:
             self._retry_failed_states_task.cancel()
-        if self._primary_peer_sync_task is not None:
```

Comment:
wouldn't we now wait for long sync before we shutdown ?

Reply:
From my understanding the underlying API call task will be anyway canceled on shutdown/disconnect via https://github.com/Chia-Network/chia-blockchain/blob/9f0220771f12549a9b566bc6e59fa10c527e8041/chia/server/ws_connection.py#L306  which should lead to cancelation of the long sync unless we suppress the cancelation somewhere what i don't think.

Reply:
maybe im missing something, in the current code we explicitly cancel _primary_peer_sync_task what would cancel the long sync now that we dont do that explicitly ?

Reply:
Its just canceled like before, but not via the API call like i suspected above because the API calls are queued in the `new_peak_queue`, so instead the long sync gets triggered in here https://github.com/Chia-Network/chia-blockchain/blob/1d4700dceb3524e7d9be2c71c986be9a03165754/chia/wallet/wallet_node.py#L977 which gets called here https://github.com/Chia-Network/chia-blockchain/blob/1d4700dceb3524e7d9be2c71c986be9a03165754/chia/wallet/wallet_node.py#L467 after a new peak message was put into the queue via an API call.

The new peak queue processing gets canceled here still https://github.com/Chia-Network/chia-blockchain/blob/1d4700dceb3524e7d9be2c71c986be9a03165754/chia/wallet/wallet_node.py#L303

Reply:
im not talking about t _process_new_subscriptions_task im talking about the call to long_sync which used to be a task but now we just do await self_long_sync() and im not sure that we break anywhere in the middle of long_sync in case of a shutdown

here https://github.com/Chia-Network/chia-blockchain/pull/14362#discussion_r1090534336

Reply:
Lets continue the conversation here, but see https://github.com/Chia-Network/chia-blockchain/pull/14362#discussion_r1090552770 and i think we might need [9cea0de](https://github.com/Chia-Network/chia-blockchain/pull/14362/commits/9cea0deca7a212999ea5cb481177166a7b36adde) to make it work unless it also raises on any other await inside this task.. maybe @altendky has some more thoughts about this but from my understanding the long sync should be canceled like before now.